src/perf/lib/Tcp.cpp

```diff
@@ -222,29 +227,48 @@ void TcpWorker::Shutdown()
     }
 }
 
-CXPLAT_THREAD_CALLBACK(TcpWorker::WorkerThread, Context)
+//
+// Runs one iteration of the worker loop. Returns FALSE when it's time to exit.
+//
+BOOLEAN
+TcpWorker::DoWork(
+    _Inout_ void* Context,
+    _Inout_ CXPLAT_EXECUTION_STATE* // State
+    )
 {
     TcpWorker* This = (TcpWorker*)Context;
-
-    while (!This->Engine->Shutdown) {
-        TcpConnection* Connection;
-        CxPlatDispatchLockAcquire(&This->Lock);
-        if (!This->Connections) {
-            Connection = nullptr;
-        } else {
-            Connection = This->Connections;
-            This->Connections = Connection->Next;
-            if (This->ConnectionsTail == &Connection->Next) {
-                This->ConnectionsTail = &This->Connections;
-            }
-            Connection->QueuedOnWorker = false;
-            Connection->Next = NULL;
+    TcpConnection* Connection;
+    if (This->Engine->Shutdown) {
+        return FALSE;
+    }
+    CxPlatDispatchLockAcquire(&This->Lock);
+    if (!This->Connections) {
+        Connection = nullptr;
+    } else {
+        Connection = This->Connections;
+        This->Connections = Connection->Next;
+        if (This->ConnectionsTail == &Connection->Next) {
+            This->ConnectionsTail = &This->Connections;
         }
-        CxPlatDispatchLockRelease(&This->Lock);
-        if (Connection) {
-            Connection->Process();
-            Connection->Release();
-        } else {
+        Connection->QueuedOnWorker = false;
+        Connection->Next = NULL;
+    }
+    CxPlatDispatchLockRelease(&This->Lock);
+    if (Connection) {
+        Connection->Process();
+        Connection->Release();
+        This->ExecutionContext.Ready = TRUE; // We just did work, let's keep this thread hot.
+    } else {
+        This->ExecutionContext.Ready = FALSE; 
```

Comment:
Don't have this `else` clause. It's unnecessary and all extra instructions are pure overhead. Since this is a perf sensitive code path, we want to remove any fluff.

Reply:
I removed the `if` check in the loop. Does having an `else` clause add more instructions in assembly than an extra `if` check?

Reply:
This function must assume `Ready == FALSE` on entry. And the WorkerThread should enforce that. That's how the CxPlat execution context works.

Reply:
done